ale/drivers/selene_drivers.py

```diff
@@ -435,3 +414,318 @@ def sensor_model_version(self):
+        -------
+        : float
+          spacecraft clock stop count in seconds
+        """
+        return self.label.get('CORRECTED_SC_CLOCK_STOP_COUNT').value
+
+    @property
+    def spacecraft_clock_start_count(self):
+        """
+        The original SC_CLOCK_START_COUNT key is often incorrect and cannot be trusted.
+        Therefore we get this information from CORRECTED_SC_CLOCK_START_COUNT
+
+        Returns
+        -------
+        : float
+          spacecraft clock start count in seconds
+        """
+        return self.label.get('CORRECTED_SC_CLOCK_START_COUNT').value
+
+    @property
+    def ephemeris_start_time(self):
+        """
+        Returns the ephemeris start time of the image. Expects spacecraft_id to
+        be defined. This should be the integer naif ID code of the spacecraft.
+
+        Returns
+        -------
+        : float
+          ephemeris start time of the image
+        """
+        return spice.sct2e(self.spacecraft_id, self.spacecraft_clock_start_count)
+
+    @property
+    def detector_center_line(self):
+        """
+        Returns the center detector line of the detector. Expects tc_id to be
+        defined. This should be a string of the form LISM_TC1 or LISM_TC2.
+
+        We subtract 0.5 from the center line because as per the IK:
+        Center of the first pixel is defined as "1.0".
+
+        Returns
+        -------
+        : int
+          The detector line of the principle point
+        """
+        return spice.gdpool('INS{}_CENTER'.format(self.ikid), 0, 2)[1] - 0.5
+
+    @property
+    def detector_center_sample(self):
+        """
+        Returns the center detector sample of the detector. Expects tc_id to be
+        defined. This should be a string of the form LISM_TC1 or LISM_TC2.
+
+        We subtract 0.5 from the center sample because as per the IK:
+        Center of the first pixel is defined as "1.0".
+
+        Returns
+        -------
+        : int
+          The detector sample of the principle point
+        """
+        return spice.gdpool('INS{}_CENTER'.format(self.ikid), 0, 2)[0] - 0.5
+
+    @property
+    def focal2pixel_samples(self):
+        """
+        Calculated using 1/pixel pitch
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          focal plane to detector samples
+        """
+        pixel_size = spice.gdpool('INS{}_PIXEL_SIZE'.format(self.ikid), 0, 1)[0]
+        return [0, 0, -1/pixel_size]
+
+
+    @property
+    def focal2pixel_lines(self):
+        """
+        Calculated using 1/pixel pitch
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          focal plane to detector lines
+        """
+        pixel_size = spice.gdpool('INS{}_PIXEL_SIZE'.format(self.ikid), 0, 1)[0]
+        return [0, 1/pixel_size, 0]
```

Comment:
From IK, X+ id down-track

Reply:
The + seems odd, not that it matters, but did they not fly both directions?

Reply:
I think they only flew in one direction and that the same part of the spacecraft was always "forward".

Reply:
Here's the ascii art from the IK
```

   The direction along the row of pixels of each band is defined to be 
   Y axis. The +Y is the direction toward the radiator side from the 
   detectors. Z axis is defined to be "ideal" boresight. 


                       Corner pixel
                      view direction 
                           ^
         Center pixel       \             X(Along-track)
            view   <.        \           ^   CCD plane
          direction  `-.      \          |
                        `-.    \         |     .^ -Y(Cross-track)
                           `-.  \        |   .'
                              `-.\    F  | .'
                     <-----------`o.-----o'  Detector line in
                   Z        Focal  \`-.  | .  the CCD plane
                            point   \  `-*'
                                     \ .'| Center pixel
                                      *  |
                                  Corner
                                   pixel           
```